### PR TITLE
Report more metrics and add output annotations

### PR DESCRIPTION
**Pull request by bibilujan**
We want to report more metrics after ichorCNA is run, previously only coverage was reported. https://jira.oicr.on.ca/browse/GRD-422 , we now want to report: 

-   mean coverage
-   total reads 
-   lanes sequenced
-   reads per lane

In addition, there is a request to sort the output pdfs to folders for easier review of the data: https://jira.oicr.on.ca/browse/GBS-3418

For this we are adding annotations to the pdf output files.

### CODE REVIEW

**Review by lheisler — APPROVED** [review]

File: ichorCNA.wdl [hunk 1]
```diff
@@ -370,6 +461,50 @@ task runIchorCNA {
...
+    plotFileType: "File format for output plots. Default: [pdf]."
+    plotYLim: "ylim to use for chromosome plots. Default: [c(-2,2)]."
+    outDir: "Output Directory. Default: [./]."
+    libdir: "Script library path."
+    modules: "Environment module name and version to load (space separated) before command execution."
+    mem: "Memory (in GB) to allocate to the job."
+    timeout: "Maximum amount of time (in hours) the task can run for."
+  }
+
   command <<<
```
Comment (mlaszloffy) [inline ichorCNA.wdl:508]: Can `set -euo pipefail` be added here?

File: ichorCNA.wdl [hunk 2]
```diff
@@ -483,7 +596,179 @@ task runIchorCNA {
...
+
+  parameter_meta {
+    inputbam: "Input bam."
+    outputFileNamePrefix: "Output prefix to prefix output file names with."
+    jobMemory: "Memory (in GB) to allocate to the job."
+    modules: "Environment module name and version to load (space separated) before command execution."
+    timeout: "Maximum amount of time (in hours) the task can run for."
+  }
+
+  command <<<
```
Comment (mlaszloffy) [inline ichorCNA.wdl:639]: Can `set -euo pipefail` be added here?

File: ichorCNA.wdl [hunk 3]
```diff
@@ -483,7 +596,179 @@ task runIchorCNA {
...
+    preBamMetrics: "pre-merge bam metrics."
+    bamMetrics: "bam metrics."
+    allSolsMetrics: "metrics for all solutions"
+    outputFileNamePrefix: "Output prefix to prefix output file names with."
+    jobMemory: "Memory (in GB) to allocate to the job."
+    modules: "Environment module name and version to load (space separated) before command execution."
+    timeout: "Maximum amount of time (in hours) the task can run for."
+  }
+
+  command <<<
```
Comment (mlaszloffy) [inline ichorCNA.wdl:690]: Can `set -euo pipefail` be added here?

**Comment by bibilujan** [comment]
@mlaszloffy the build was failing because there was a pipefail at the read_count calculation, not a "real" failure, but based on what I could understand maybe something similar to this https://stackoverflow.com/questions/41516177/bash-zcat-head-causes-pipefail where head get the output before the first command is done. Changed it to grep and get the same result and no pipefail. 

A good catch was that read_counts was being calculated differently on premerge bams and post, so I have updated this to the new command.

**Review by mlaszloffy — APPROVED** [review]

**Pull request merged**